opentelemetry/src/sdk/trace/span.rs

```diff
@@ -51,11 +53,19 @@ impl Span {
         span_context: SpanContext,
         data: Option<SpanData>,
         tracer: sdk::trace::Tracer,
+        span_limit: Option<SpanLimits>,
```

Comment:
Does this need to be optional? Could switch tests to use `Default::default()` instead of `None`

Reply:
If the `span_limit` is `None`. Then we will try to get the span limit from the tracer provider. If there isn't a tracer provider connected with the tracer, we will use the default. 

Sometimes we may change the `span_limit` during the span building(For example, we may record the number of dropped entities) so we may pass modified span_limit. But we want to keep the span limit the same as the tracer provider, we can pass `None`.

Reply:
Hm yeah we should store them somewhere. Maybe best path is to have `SpanLimits` not be optional here and add dropped counts to `SpanData`? keeps things simple and tracks attribute drops without having to upgrade the provider.

Reply:
Sounds reasonable. I removed the option around `SpanLimits`. I probably would open another PR to address how to record and expose the number of dropped entities.